### ISSUE TITLE
type assignment error when assigning onload handler

### ISSUE THREAD

**Issue by nicks** [google/closure-compiler#1454]
Code:

``` js
var reader = new FileReader()
reader.onload = function (e) {
  var array = new Uint8Array(e.target.result)
}
```

Expected: no warnings
Actual:

```
input0:3: WARNING - The right side in the assignment is not a subtype of the left side.
left side  : function(ProgressEvent):?|null
right side : function({target:{result:Array<number>|ArrayBuffer|ArrayBufferView|null|number} (loose)} (loose)):undefined

    reader.onload = function (e) {
    ^
```

http://closure-compiler-debugger.appspot.com/#input0%3D%250A%2520%2520%2520%2520var%2520reader%2520%253D%2520new%2520FileReader()%250A%2520%2520%2520%2520reader.onload%2520%253D%2520function%2520(e)%2520%257B%250Anew%2520Uint8Array(e.target.result)%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by nicks** [comment]
(with new type inference)

**Comment by dimvar** [comment]
I think it complains because result is not defined on EventTarget, which seems right to me. Which type is result defined on here?

**Comment by nicks** [comment]
e.target should be a {FileReader}, and e.target.result should be a {string|Blob|ArrayBuffer}.

even if we agree that there should be a warning, i'm not sure that the current warning message is defensible.

**Comment by dimvar** [comment]
The error message isn't great, but it's the best NTI can do in this case. It complains that the type of e is not a supertype of ProgressEvent, which is needed for the rhs to be a subtype of the lhs.

What if you annotate e with your desired type (I'm not sure what it is), does that fix the issue?

**Comment by nicks** [comment]
e is a ProgressEvent. The compiler agrees.

arguably, the problem is recognizing what e.target is

**Comment by dimvar** [comment]
Does NTI still complain if you annotate e to be a ProgressEvent?
